github/github.go

```diff
@@ -396,7 +396,10 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Res
 
 	// If we've hit rate limit, don't make further requests before Reset time.
 	if err := c.checkRateLimitBeforeDo(req, rateLimitCategory); err != nil {
-		return nil, err
+		return &Response{
+			Response: err.Response,
```

Comment:
I have one potential concern about this change.

I think it's somewhat unfortunate that we ended up being forced to create a fake/mock `*http.Response` even in situations where no remote HTTP request was actually made. I am referring to [this](https://github.com/google/go-github/blob/9be6ce168fa69a3cc2db27a776950a42d18c2ae0/github/github.go#L465-L472).

It was originally done in #347 because we (or I) didn't want to break clients that assumed `*http.Response` was never `nil`, since it used to never be `nil` in prior to that change.

So I think it's somewhat disadvantageous that we would now be propagating the fake `*http.Response` in more situations.

However, even if we had control over this, I'm not sure what's better. If a remote API request is not made (because we've predicted it'll fail due to exceeded rate limit), should the client be able to tell it apart from when a remote API request was made? It wasn't documented in the past, but one could've done that by checking if `resp` was nil. This changes that.

I'm pointing out the above because I wanted to think it through. But I don't think it's a reason to avoid going with this change. This change actually increases consistency (in that we return `resp` and `resp.Rate` in more cases).

So, I think I'm okay with doing this, but I want to take another day to think it over. Meanwhile, I'd love to hear your perspective on the above @gmlewis.